openfecli/commands/gather.py

```diff
@@ -121,31 +122,28 @@ def _get_names(result: dict) -> tuple[str, str]:
     tuple[str, str]
         Ligand names corresponding to the results.
     """
-    try:
-        nm = list(result["unit_results"].values())[0]["name"]
-
-    except KeyError:
-        raise ValueError("Failed to guess names")
 
-    # TODO: make this more robust by pulling names from inputs.state[A/B].name
+    # TODO: I don't like this [0][0] indexing, but I can't think of a better way currently
+    protocol_data = list(result["protocol_result"]["data"].values())[0][0]
+    ligand_A = gufe.SmallMoleculeComponent.from_dict(
+        protocol_data["inputs"]["stateA"]["components"]["ligand"]
+    )
+    ligand_B = gufe.SmallMoleculeComponent.from_dict(
+        protocol_data["inputs"]["stateB"]["components"]["ligand"]
+    )
 
-    toks = nm.split()
-    if toks[2] == "repeat":
-        return toks[0], toks[1]
-    else:
-        return toks[0], toks[2]
+    return ligand_A.name, ligand_B.name
 
 
-def _get_type(res: dict) -> Literal["vacuum", "solvent", "complex"]:
+def _get_type(result: dict) -> Literal["vacuum", "solvent", "complex"]:
     """Determine the simulation type based on the component names."""
-    # TODO: use component *types* instead here
-    list_of_pur = list(res["protocol_result"]["data"].values())[0]
-    pur = list_of_pur[0]
-    components = pur["inputs"]["stateA"]["components"]
 
-    if "solvent" not in components:
+    protocol_data = list(result["protocol_result"]["data"].values())[0][0]
+    chem_sys_A = gufe.tokenization.GufeTokenizable.from_dict(protocol_data["inputs"]["stateA"])
```

Comment:
[nit] Why not use the ChemicalSystem object directly?

Reply:
ah, we can! that was an artifact from me trying some different things during development. 
fixed!